bin/board.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class Board
+  attr_reader :winner
+  attr_reader :has_movs
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @blocks = Array.new(9, ' ')
+    @flag = true
+    @winner = nil
+    @has_movs = true
+
+    @winner_positions = [
+      [0, 1, 2], [3, 4, 5], [6, 7, 8],
+      [0, 3, 6], [1, 4, 7], [2, 5, 8],
+      [0, 4, 8], [2, 4, 6]
+    ]
+  end
+
+  # Method collecting the names of the player
+  def ask_names
+    [@player1, @player2].each_with_index do |ply, index|
+      system 'clear'
+      puts 'Insert player' + (index + 1).to_s + "'s name"
+      name = gets.chomp
+      ply.name = name
+    end
+  end
```

Comment:
**[REQUIRED]**
* All the user interaction code should reside in the `main.rb` file. Please read carefully on the requirement for milestone 3 [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/). `puts/print` and `get.chomp` should reside in `bin/main.rb` only.

Reply:
moved to main.rb
```
mplayer1 = Player.new('X')
mplayer2 = Player.new('O')
my_board = Board.new(mplayer1, mplayer2)

play_again = true

# collecting names of the players
[mplayer1, mplayer2].each_with_index do |ply, index|
  system 'clear'
  puts 'Insert player' + (index + 1).to_s + "'s name"
  name = gets.chomp
  ply.name = name
end